### PR TITLE
Arxivce 1273 new base

### PR DESCRIPTION
**Pull request by mnazzaro**
There are a ton of changes here. 

To be clear, base is not done, but this is a good intermediate step that I'm confident, after rigorous testing, shouldn't break anything. 

Some highlights:
- All db models moved to base
- Most of domain moved to base (biggest thing here is DocMetadata)
- Really good type hints now since the db is type mapped
- No more Flask-SQLAlchemy
- Upgrade to SQLAlchemy 2.*
- Upgrade to python 3.11
- Upgrade to Flask 3.*

One crucial change before this is deployed is to change the BROWSE_SQLALCHEMY_DB_URI environment variable to be called CLASSIC_DB_URI in cloud run.

### CODE REVIEW

File: browse/controllers/archive_page/__init__.py [hunk 1]
```diff
@@ -4,7 +4,13 @@
 from typing import Any, Dict, List, Tuple, Optional
 from http import HTTPStatus as status
 
-from arxiv.taxonomy.definitions import ARCHIVES, ARCHIVES_SUBSUMED, CATEGORIES
+from arxiv.taxonomy.definitions import (
+    ARCHIVES, 
+    ARCHIVES_SUBSUMED, 
+    CATEGORIES,
+    tCategory,
+    tArchive
```
Comment (kyokukou) [inline browse/controllers/archive_page/__init__.py:12]: nitpick- tArchive and tCategory names feel confusing

File: browse/controllers/list_page/__init__.py [hunk 2]
```diff
@@ -131,7 +131,8 @@ def get_listing(subject_or_category: str,
        Number of articles to show
     """
     
-
+    logging.warning(f"{current_app.config['DOCUMENT_LISTING_SERVICE']}: {type(get_listing_service())}")
```
Comment (kyokukou) [inline browse/controllers/list_page/__init__.py:134]: are these warning worthy?

Comment (mnazzaro) [reply browse/controllers/list_page/__init__.py:134]: Oh oops nope that's extra logging

File: browse/controllers/archive_page/__init__.py (hunk 1 above)
Comment (mnazzaro) [reply browse/controllers/archive_page/__init__.py:12]: Suggestions? They're TypedDicts

**Comment by mnazzaro** [comment]
Some things I'm going to change today:

- Make browse inherit base settings or something like that
- Make sure everything I moved to base is deleted in browse
- Move tests for stuff I put in base over to base
- Move object store and file manipulation over to base (I think I'm going to do this part in a branch off of this one)

File: browse/routes/ui.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 )
 from browse.controllers.openurl_cookie import make_openurl_cookie, get_openurl_page
 from browse.controllers.cookies import get_cookies_page, cookies_to_set
-from browse.exceptions import AbsNotFound
+from browse.exceptions import AbsNotFound # TODO: BASE
```
Comment (kyokukou) [inline browse/routes/ui.py:37]: did you miss this one?

File: browse/services/database/__init__.py [hunk 4]
```diff
@@ -170,94 +171,97 @@ def get_recent_trackback_pings(max_trackbacks: int = 25) \
...
-    return dt
+    if timestamp:
+        dt = datetime.fromtimestamp(timestamp, tz=tz)
+        dt = dt.astimezone(tz=tzutc())
+        return dt
+    return None
 
 
 # used on abs page
 @db_handle_error(db_logger=logger, default_return_val=0)
```
Comment (kyokukou) [inline browse/services/database/__init__.py:202]: I remember Brian C saying something about not handling errors and letting flask do it. I know these were already here, but I wonder if the refactor would be a good time to look into it

File: browse/services/database/__init__.py [hunk 5]
```diff
@@ -351,22 +354,29 @@ def get_hourly_stats_count(stats_date: Optional[date]) -> Tuple[int, int, int]:
     normal_count = 0
     admin_count = 0
     num_nodes = 0
-    rows = (
-        db.session.query(
-            func.sum(stats_hourly.c.connections).label("num_connections"),
-            stats_hourly.c.access_type,
-            func.max(stats_hourly.c.node_num).label("num_nodes"),
+    print ("MADE IT HERE")
```
[the remark this answers is not in the document]
Comment (mnazzaro) [reply browse/services/database/__init__.py:357]: Lol nope

File: browse/controllers/list_page/__init__.py [hunk 6]
```diff
@@ -51,17 +51,17 @@
...
 from browse.services.documents import get_doc_service
-from browse.services.documents.format_codes import formats_from_source_flag
+from arxiv.formats import formats_from_source_flag
 from browse.services.listing import (Listing, ListingNew, NotModifiedResponse,
                                      get_listing_service, ListingItem)
 
 from browse.formatting.latexml import get_latexml_url
 
-from flask import request, url_for, redirect
+from flask import request, url_for, redirect, current_app
```
Comment (bdc34) [inline browse/controllers/list_page/__init__.py:64]: `current_app` is added to the imports but I don't see any other code changes, is it not used?

File: browse/factory.py [hunk 7]
```diff
@@ -76,12 +73,12 @@ def create_web_app(**kwargs) -> Flask: # type: ignore
     app.jinja_env.filters['arxiv_id_doi_filter'] = urlizer(['arxiv_id', 'doi'])
     app.jinja_env.filters['tidy_filesize'] = tidy_filesize
 
-    @app.before_first_request
-    def check_services()->None:
+    with app.app_context():
```
Comment (bdc34) [inline browse/factory.py:76]: :+1:

File: browse/services/documents/db_implementation/db_abs.py [hunk 8]
```diff
@@ -92,32 +95,34 @@ def _to_docmeta(all_versions: List[Metadata], latest: Metadata, ver_of_interest:
...
     modified = ver_of_interest.updated or ver_of_interest.created
-    modified = modified.replace(tzinfo=timezone.utc)  # Verified as UTC in DB
+    modified = modified.replace(tzinfo=timezone.utc) # type: ignore
+    # ^verified as UTC in DB
     primary_category, secondary_categories, primary_archive = _classification_for_metadata(identifier, latest)
 
     this_version = DocMetadata(
         raw_safe='',
-        abstract=ver_of_interest.abstract,
+        abstract=ver_of_interest.abstract, # type: ignore
```
Comment (bdc34) [inline browse/services/documents/db_implementation/db_abs.py:116]: What causes these type ignores? Are these cases where the DB can be null but the model is not Optional?

File: browse/services/documents/fs_implementation/legacy_fs_paths.py [hunk 9]
```diff
@@ -2,12 +2,11 @@
 
 from typing import Optional, List, Dict
 
-from browse.services.anypath import to_anypath, APath
+from arxiv.files.anypath import to_anypath, APath
```
Comment (bdc34) [inline browse/services/documents/fs_implementation/legacy_fs_paths.py:5]: fs_implementation was written before the ObjectStore stuff. 

In the future we could get rid of cloudlibpath and anypath and just use ObjectStore.

File: tests/dissemination/test_integration.py [hunk 10]
```diff
@@ -337,9 +337,9 @@ def test_html_src(host):
     """Submissions with HTML source"""
     # legacy returns 200 with msg: "Unavailable, The author has provided no source to generate PDF, and no PDF."
     resp = requests.get(f"{host}/pdf/cond-mat/9906075v1")
-    assert resp.status_code == 404
+    assert resp.status_code == 500
```
Comment (bdc34) [inline tests/dissemination/test_integration.py:340]: This should be a 404.

Comment (mnazzaro) [reply tests/dissemination/test_integration.py:340]: Oops ya

**Review by bdc34 — CHANGES_REQUESTED** [review]
One comment about the 404 vs 500 status of src not found needs to be addressed. Otherwise, it looks good.

File: browse/services/documents/db_implementation/db_abs.py (hunk 8 above)
Comment (mnazzaro) [reply browse/services/documents/db_implementation/db_abs.py:110]: Yes exactly

File: tests/dissemination/test_integration.py (hunk 10 above)
Comment (bdc34) [reply tests/dissemination/test_integration.py]: Actually i see I'm wrong, this is a case where we really expect there to be a PDF but there isn't one. 500 is fine.

**Review by bdc34 — APPROVED** [review]

File: browse/services/database/listings.py [hunk 11]
```diff
@@ -347,21 +353,21 @@ def _metadata_to_listing_item(meta: Metadata, type: AnnounceTypes) -> ListingIte
...
+        modified=datetime.fromtimestamp(float(modtime))
+    elif updated is not None and modtime is None:
         modified=updated
     doc = DocMetadata(  
         arxiv_id=meta.paper_id,
         arxiv_id_v=f"{meta.paper_id}v{meta.version}",
-        title=meta.title,
-        authors=AuthorList(meta.authors),
-        abstract=meta.abstract,
+        title=meta.title, # type: ignore
```
Comment (kyokukou) [inline browse/services/database/listings.py:364]: why do some of these have to be type ignored?

File: browse/services/listing/__init__.py [hunk 12]
```diff
@@ -9,14 +9,15 @@
...
-from browse.domain.metadata import DocMetadata
+from arxiv.document.metadata import DocMetadata
 from browse.services import HasStatus
 
 
 def get_listing_service() -> "ListingService":
     """Get the listing service configured for the app context."""
     if "listing_service" not in g:
         fn = current_app.config["DOCUMENT_LISTING_SERVICE"]
+        print (type(fn))
```
Comment (kyokukou) [inline browse/services/listing/__init__.py:20]: print statement

File: browse/services/listing/__init__.py [hunk 13]
```diff
@@ -28,10 +29,10 @@ def fs_listing(config: dict, _: Any) -> "ListingService":
...
 
 def db_listing(config: dict, _: Any) -> "ListingService":
-    """Factory function for filesystem-based listing service."""
-    from browse.services.database import models
-    from .db_listing import DBListingService
-    return DBListingService(models.db)
+    """Factory function for DB backed listing service."""
+    from .db_listings import DBListingService
+    # maybe pass in the specific classes for the tables we need?
+    return DBListingService()
```
Comment (kyokukou) [inline browse/services/listing/__init__.py:35]: did you test using the db listing service without it being passed anything?

File: tests/data/latexml.db
Comment (kyokukou) [inline tests/data/latexml.db:1]: did you mean to add a megabyte to this?

Comment (kyokukou) [reply tests/data/latexml.db:1]: maybe check if the tests work with this change reverted? I had to manually untrack this to stop accidently adding to it

File: tests/test_year_page.py [hunk 14]
```diff
@@ -77,7 +77,7 @@ def test_get_yearly_article_counts(app_with_db):
         # 2007 mid id-swap
         # TODO cant test early 2007 data on sqlite
 
-@patch("browse.services.listing.db_listing.get_yearly_article_counts")
+@patch("browse.services.listing.db_listings.get_yearly_article_counts")
```
Comment (kyokukou) [inline tests/test_year_page.py:80]: I dont see this file being renamed in your changes

**Review by kyokukou — CHANGES_REQUESTED** [review]
look at the print statements, logger warning and ever growing latexml.db. The others comments are more questions/suggestions

**Comment by kyokukou** [comment]
also, is this built on a version of base that doesn't pass bases tests?

File: browse/services/database/__init__.py [hunk 15]
```diff
@@ -538,25 +539,21 @@ def get_user_display_name(user_id: int) -> Optional[str]:
...
+        select(Document, t_arXiv_paper_owners)
+        .filter(Document.document_id == t_arXiv_paper_owners.c.document_id)
+        .filter(t_arXiv_paper_owners.c.user_id == user_id)
+        .filter(t_arXiv_paper_owners.c.flag_author == 1)
+        .filter(t_arXiv_paper_owners.c.valid == 1)
         .filter(Document.paper_id.notlike('test%'))
         .order_by(Document.dated.desc())
-        .all()
-    )
+    ).scalars().all()
```
[the remark this answers is not in the document]
Comment (mnazzaro) [reply browse/services/database/__init__.py:557]: scalars returns the stuff that would be inside the row like execute returns. So they could be full ORM objects, they just won't be in a row

File: browse/services/listing/__init__.py (hunk 13 above)
Comment (mnazzaro) [reply browse/services/listing/__init__.py:34]: Yup!

File: tests/data/latexml.db
Comment (mnazzaro) [reply tests/data/latexml.db:1]: I don't know why the .gitignore isn't pick up on it. How'd you go about not tracking it?

File: browse/services/listing/__init__.py (hunk 13 above)
Comment (kyokukou) [reply browse/services/listing/__init__.py:34]: sweet

**Review by kyokukou — APPROVED** [review]

**Pull request merged**